mantidproject/mantid
# ISIS Powder PEARL enable absorption corrections for non-vanadium samples

Previously, absorption corrections were not enabled for PEARL when processing anything other than vanadium run. The absorption correction routines were exposed for all samples and two new parameters were added to `isis_powder` to support generating and saving correction factors. A system test was also added to monitor the behaviour of focusing with all corrections except absorption disabled.

**To test:**

```
from isis_powder import Pearl

pearl = Pearl(config_file=r"\\olympic\Babylon5\Public\JoeRamsay\PEARL\pearl_shared_config.yaml")

no_corrections = pearl.focus(run_number=99489, do_absorb_corrections=False)
RenameWorkspace(InputWorkspace=no_corrections, OutputWorkspace="no_corrections")
RenameWorkspaces(InputWorkspaces=no_corrections.getNames(), Suffix="_no_corr")

absorb_corr_file = r"\\olympic\Babylon5\Public\JoeRamsay\PEARL\pearl_99589_corr.nxs"
generated_corrections = pearl.focus(run_number=99489, do_absorb_corrections=True,
                                    generate_absorb_corrections=True,
                                    absorb_corrections_out_filename=absorb_corr_file)
RenameWorkspace(InputWorkspace=generated_corrections, OutputWorkspace="gen_corrections")
RenameWorkspaces(InputWorkspaces=generated_corrections.getNames(), Suffix="_gen_corr")

default_corrections = pearl.focus(run_number=99489, do_absorb_corrections=True,
                                  generate_absorb_corrections=False)
RenameWorkspace(InputWorkspace=default_corrections, OutputWorkspace="default_corrections")
RenameWorkspaces(InputWorkspaces=default_corrections.getNames(), Suffix="_def_corr")
```

Make sure this script runs without any problems (it will take around 15 minutes to run due to the `AbsorptionCorrection` algorithm).

The script focused run 99489 3 ways:
1. No corrections applied
2. Absorption correction factors are generated from the workspace and applied
3. The default PEARL correction factors are applied

Make sure `\\olympic\Babylon5\Public\JoeRamsay\PEARL\pearl_99589_corr.nxs` has been generated, then inspect the data. Plotting bank 1 from each of the generated group workspaces should give you something like this, where the lowest number of counts is for no corrections, and the highest number is for the default corrections, with the generated corrections sitting slightly below the default.
![corrections](https://user-images.githubusercontent.com/29916298/33937055-c9133a94-dff9-11e7-9af9-ef898a7eff03.PNG)

Fixes #20914 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/8726575e1cdec3d050d2de006472c2b23ce4e868)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
